v2.2/query-behavior-troubleshooting.md

```diff
@@ -1,31 +1,94 @@
+
+CPU usage: [CPU percent](https://www.cockroachlabs.com/docs/v2.1/admin-ui-hardware-dashboard.html#cpu-percent)
+
+Network latency: [Service latency SQL 99th percentile](https://www.cockroachlabs.com/docs/v2.1/admin-ui-overview-dashboard.html#service-latency-sql-99th-percentile)
+
+## Single hot node: One node has much higher resource usage than other nodes
+
+To determine if you have a hot node in your cluster, [access the Admin UI](admin-ui-access-and-navigate.html#access-the-admin-ui), click **Metrics** on the left, and navigate to the following graphs. Hover over each of the following graphs to see the per-node values of the metrics. If one of the nodes has a higher value, you have a hot node in your cluster.
+
+-   Replication dashboard > Average queries per store graph.
+
+-   Overview Dashboard > Service Latency graph
+
+-   Hardware Dashboard > CPU percent graph
+
+-   SQL Dashboard > SQL Connections graph
+
+-   Hardware Dashboard > Disk IOPS in Progress graph
+
+**Solution:**
+
+-   If you have a small table that fits into one range, then only one of the nodes will be used. This is expected behavior.
+
+-   If the **SQL Connections** graph shows that one node has a higher number of SQL connections and other nodes have zero connections, check if your app is set to talk to only one node.
+
+-   Check load balancer settings.
+
+-   Check for [transaction contention](https://www.cockroachlabs.com/docs/stable/performance-best-practices-overview.html#understanding-and-avoiding-transaction-contention).
+
+-   Use `UUID` instead of `SERIAL`. See [UUID Best practices](https://www.cockroachlabs.com/docs/stable/performance-best-practices-overview.html#unique-id-best-practices) for more information.
+
+## INSERT/UPDATE statements are slow
+
+Use the [Statements page](https://www.cockroachlabs.com/docs/stable/admin-ui-statements-page.html) to identify the slow [SQL statements](https://www.cockroachlabs.com/docs/stable/sql-statements.html). To view the Statements page, [access the Admin UI](https://www.cockroachlabs.com/docs/stable/admin-ui-access-and-navigate.html#access-the-admin-ui) and then click Statements on the left.
+
+Refer to the following documents to improve `INSERT` / `UPDATE` performance:
+
+-   [Multi-row DML](https://www.cockroachlabs.com/docs/stable/performance-best-practices-overview.html#multi-row-dml-best-practices)
+
+-   [Bulk-Insert best practices](https://www.cockroachlabs.com/docs/stable/performance-best-practices-overview.html#bulk-insert-best-practices)
+
+## Per-node queries per second (QPS) is high
+
+If a cluster is not idle, it is useful to monitor the per-node queries per second. Cockroach will automatically distribute load throughout the cluster. If one or more nodes is not performing any queries there is likely something to investigate. See `exec_success` and `exec_errors` which track operations at the KV layer and `sql_{select,insert,update,delete}_count` which track operations at the SQL layer.
```

Comment:
I'd remove "If a cluster is not idle"

Reply:
Umm..do we want them to monitor QPS even when the cluster is idle? 🤔